src/Features/Core/Portable/CodeFixes/CodeFixService.cs

```diff
@@ -758,6 +760,17 @@ private async Task<IEnumerable<Diagnostic>> GetDocumentDiagnosticsAsync(Document
             return await diagnostics.ToDiagnosticsAsync(document.Project, cancellationToken).ConfigureAwait(false);
         }
 
+        private async Task<IEnumerable<Diagnostic>> GetDocumentSpanDiagnosticsAsync(Document document, TextSpan span, ImmutableHashSet<string>? diagnosticIds, bool includeSuppressedDiagnostics, CancellationToken cancellationToken)
```

Comment:
Shouldn't this method be called by the one above and just pass in the span of the document?

Reply:
No, the IDE DiagnosticAnalyzerService uses different diagnostic getters for span based diagnostic queries and full document diagnostic queries, each of which is optimized for span vs full-span scenarios. [LatestDiagnosticsForSpanGetter](https://sourceroslyn.io/#Microsoft.CodeAnalysis.Features/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer_GetDiagnosticsForSpan.cs,55) for span based queries and [IdeLatestDiagnosticGetter](https://sourceroslyn.io/#Microsoft.CodeAnalysis.Features/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer_GetDiagnostics.cs,35) for full document/project/solution queries.